replica/replica.go

```diff
@@ -205,6 +205,15 @@ func (replica *Replica) TimeoutPrecommit(ctx context.Context, timeout timer.Time
 	}
 }
 
+// UpdateState updates the state of the replica's underlying process with the
+// given height, round and step
+func (replica *Replica) UpdateState(height process.Height, round process.Round, step process.Step) {
```

Comment:
I do not support the addition of this method in the way that it has been defined. It is dangerous to allow the process state to be arbitrarily set. The consensus algorithm makes lots of assumptions about how a process progress between rounds and steps, and I am not certain that this kind of forced state change does not violate those assumptions.

For example, to progress to a particular round/step in the expected way, the process state will have message logs and once flags set. With this method, none of that happens.

Reply:
This is not a part of the audit fixes.

I needed a way to update the state of the replica once block syncing is complete (for Archaeon). On fetching the missing blocks, I wanted to update the height of the replica process.

Now that I think of it, we won't need to update the `round` or `step`, as they will be updated by replaying the message log. But updating the height is still required.